### PR TITLE
2019 Maintenance & Transition to Gatsby Theme

### LINKED ISSUE #44: Cards
The current version of the page occupies by default the whole width of the page, making the cards too wide to scan through and leaving too much space between them.

I managed to reduce the width of the cards without squeezing them too tight, and you can refer to this screenshot:

image

I wanted to change the background color of the page to light gray since white cards with drop shadows does not look good on the white background, but a gray background does not cope well with the beige banner. I am still working on that, and I shall come with some other solutions later.

If you are happy with this, I can just put this on.

### PR DESCRIPTION
**Pull request by keplersj**
Looks like it's been about a year since this site transitioned to Gatsby, per [`d8cf9f`](https://github.com/AtomLinter/atomlinter.github.io/commit/d8cf9f06ca056a62b3748731fca893ec089f8ae3)! Seems to me it's about time for some yearly maintenance.

# Visual TL;DR

| Before | After |
| --- | --- |
| Screen Shot 2019-10-07 at 9 17 49 PM | Screen Shot 2019-10-07 at 9 17 40 PM |

# Summary of Changes

I tried my best to make sure the commits in this PR were [atomic]. The benefit of this is that it'll make it very easy to see how we got from A to B, can revert unwanted changes, and can easily summarize the changes below.

## Transition from using Static Query Higher Order Components to Hooks

In the year since this site transitioned to Gatsby, they have introduced a new [`useStaticQuery`](https://www.gatsbyjs.org/blog/2019-02-20-introducing-use-static-query/) [React Hook](https://reactjs.org/docs/hooks-intro.html) that can be used in place of the StaticQuery higher order component. In short React Hooks provide many of the same advantages as an HOC, while reducing some of the HOC boilerplate allowing for more flexibility and easier maintainability in components.

## Use Emotion to Custom Styled Components

Previously the site was using a global stylesheet to customize a few of the Material Design Lite components on the site. Using [emotion](https://emotion.sh/docs/introduction) isolates the custom styles to the components they're customizing.

## Define Navigation Bar Links to Site Metadata

Previously links in the nav bar were hard coded into the NavBar components. This PR moves the nav bar links to an array in `gatsby-config.js` allowing them to be easily customized and updated.

## Transition to Material-UI from Material Design Lite

Previously this site was using Material Design Lite as a base style framework. This was convenient for transitioning to Gatsby from Jekyll, but MDL comes with several disadvantages for Gatsby. MDL is difficult to bundle during Gatsby's initial SSR render, and introduces many render bugs when the pages rehydrates in the browser. As well MDL is design for simpler sites, not using a component framework like React. Transitioning to Material-UI solves all of these issues.

## Transition to a Gatsby Theme

And finally the big one, this PR transitions the site to a new [Gatsby Theme](https://www.gatsbyjs.org/docs/themes/), [`gatsby-theme-atom-service`](https://github.com/keplersj/gatsby-theme-atom-service). Themes are a powerful new feature of Gatsby allowing sites to abstract entires portions of their codebase into versioned packages that can be reused else where.  For this code base, this means all this repo has to worry about is the dataset it maintains and the Gatsby configuration. As well, it allows other code bases that are based on this site (like [AtomBuild](https://atombuild.github.io)) to use the same theme and stay up-to-date with site updates.

---

Closes #44

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,22 +24,11 @@
...
-    "gatsby-plugin-typescript": "2.0.13",
-    "gatsby-source-filesystem": "2.0.32",
-    "gatsby-transformer-yaml": "2.1.12",
-    "lodash": "4.17.11",
-    "material-design-lite": "1.3.0",
-    "react": "16.8.6",
-    "react-dom": "16.8.6",
-    "react-helmet": "5.2.0",
-    "typescript": "3.5.1"
+    "gatsby": "^2.15.29",
```
Comment (Arcanemagus) [inline package.json]: These should just be pinned, that way they can continue to be kept current by the dependency management tool. With the `package-lock.json` in place a range is largely useless as far as keeping things current.

Comment (keplersj) [reply package.json]: Sounds good, consider it done. Honestly, I just used `npm install ...` and it defaults to ranged versions.

**Comment by keplersj** [comment]
I've updated this Pull Request to use my new tool for Gatsby, [Carraway](https://github.com/keplersj/carraway). You can see it in use in the repo for [AtomBuild's website](https://github.com/AtomBuild/atombuild.github.io) and [my surname site](https://github.com/keplersj/surname-site).

**Comment by keplersj** [comment]
If it makes it easier to maintain I can add a [GitHub Actions config that pushes the built site to GitHub Pages](https://github.com/marketplace/actions/github-pages) to this pull request.

**Comment by Arcanemagus** [comment]
I've fixed the `GITHUB_TOKEN` in Travis CI, although it looks like `master` is currently broken, the branch here appears to be working locally.

> If it makes it easier to maintain I can add a GitHub Actions config that pushes the built site to GitHub Pages to this pull request.

That would be awesome if we could remove the dependency on TravisCI, although I'm not sure how worth it it is currently as it seems the token those jobs gets doesn't currently have sufficient rights for public repositories to deploy the page.

**Comment by Arcanemagus** [comment]
I'll wait to see if you want to setup the GitHub Actions together, otherwise I'll merge this in a few days if I don't hear back. Thanks for putting this together!

**Comment by keplersj** [comment]
Set up GitHub Actions! I have the same set up working on the [AtomBuild website's repo](https://github.com/AtomBuild/atombuild.github.io/actions), if you'd like to see it in _action_.

**Comment by Arcanemagus** [comment]
Added a `GITHUB_PAT` token with `public_repo` scope, let's see if this works.

**Comment by Arcanemagus** [comment]
A reload without cache later, and it's all updated on the live site! 🎉

**Pull request merged**